### PR TITLE
Tic Tac Toe Milestone 3

### PR DESCRIPTION
**Pull request by Tahirbhalli**
Ruby
Game logic +user interface complete project
ruby bin/main.rb

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,54 @@
...
+  This project contains:
+  This is a console base Game build in ruby. And object oriented programing concepts is used in the project
+# What it does
+Two players needed to play thi game.
+
+## Built With
+- Ruby
+
+# Start the Game 
+-bin/main.rb is the executable file which will start the Game
```
Comment (MauricioRobayo) [inline README.md]: You're still missing a space here 😅.

**Review by MauricioRobayo — CHANGES_REQUESTED** [review]
Very nice project!

It looks great and it's coming into place, but we need to fix some issues before continuing:

* [x] All classes are kept in `lib` directory ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)).
* [x] Remember that `main.rb` should only be responsible for user interaction.
* [x] Check for invalid moves. At this point, no matter what I enter for a move (text, random number, an already taken position), there is no feedback that it is an invalid move and the game seems to continue, which is not the best feedback for the user and is really confusing.
* [x] When there is a winner, please provide better feedback, currently, it just says `game over`. It should say who is the winner:
image
* [x] When there is a draw, it should stop the game and provide feedback, currently, it asks for a new move:
image

Status: __Request changes__.

File: README.md (hunk 1 above)
Comment (Tahirbhalli) [reply README.md]: done

**Review by pelzolga123 — CHANGES_REQUESTED** [review]
Project requires changes.

**Comment by pelzolga123** [comment]
## Hi Team!  
### Great job so far :+1: 
Just a few things require fixing before this project can be approved:
* [x] Game is not starting after pressing `Y` in `do you want to play again (Y/N)`, that should be fixed.
* [x] No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

### [OPTIONAL]
 * It's better to create a separate file for each class and not place them all into one.

### Please kindly fix the above issues and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

**Comment by Tahirbhalli** [comment]
> ## Hi Team! 
> ### Great job so far 
> Just a few things require fixing before this project can be approved:
> 
> * [x]  Game is not starting after pressing `Y` in `do you want to play again (Y/N)`, that should be fixed.
> * [x]  No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
> 
> ### [OPTIONAL]
> * It's better to create a separate file for each class and not place them all into one.
> 
> ### Please kindly fix the above issues and send another code review request.
> If you have any questions or need help, you can message me on slack (@ Olga Pelts). _Happy coding_

done

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team      
#### Great Job so far, but there are a few more things required. 
### General
- [ ]  No puts/print and get.chomp outside main.rb file,i can see several `puts print & gets.chomp` in `player.rb`
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by pelzolga123 — APPROVED** [review]
Approved.

**Comment by pelzolga123** [comment]
## Great  work, Congratulation :+1: Milestone Approved.
You did all changes required previously. Go ahead and merge to development.
_Goodluck in your next milestone_ *Happy coding*

**Pull request merged**